### PR TITLE
Oasis DCGAN - 45445657

### PR DESCRIPTION
**Pull request by yurisasc**
Hi Shakes,
I'm doing DCGAN for OASIS dataset (problem 6).

This pull request contains 3 scripts:
1. _dataset_loader.py_ to load the dataset
2. _model.py_ to construct and train the model
3. _main.py_ to setup and execute all the processes

For more details and usage guide, please read the README.

Author:
Yuris Cakranegara - 45445657

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages
- Code valid for Hard Difficulty
- Good results 

Other Feedback
- Overall pretty good

**Comment by rexsw** [comment]
Code looks good for hard difficulty

For next time;

Comments such as 
        # Compile the GAN model
         self.model.compile(loss="binary_crossentropy", optimizer=optimizer)
Add nothing to the readability of the code and should be avoided. Overall comments were well done

File: recognition/Yuris Cakranegara-DCGAN/dataset_loader.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import os.path
```
Comment (shakes76) [inline recognition/Yuris Cakranegara-DCGAN/dataset_loader.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.
